### DISCUSSION TITLE
dockerfile Error！！！

### DISCUSSION THREAD

**Discussion by drjkris** [mitmproxy/mitmproxy#5668] [Q&A]
Requirement already satisfied: wheel in /usr/local/lib/python3.10/site-packages (0.37.1)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: pip install --upgrade pip
ERROR: Directory '/wheels/' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: pip install --upgrade pip
The command '/bin/sh -c pip install wheel && pip wheel --wheel-dir /wheels /wheels/${MITMPROXY_WHEEL}' returned a non-zero code: 1

**Comment by mhils** [comment]
You need to follow the instructions in the README at https://github.com/mitmproxy/mitmproxy/tree/main/release/docker.

**Reply by fluidum** [reply]
@mhils is there some pipeline that triggers refered Dockerfile directly and that is why VERSION and MITMPROXY_WHEEL arguments are not predefined?

**Reply by mhils** [reply]
@noraisk: https://github.com/mitmproxy/mitmproxy/blob/main/release/build-and-deploy-docker.py
